### PR TITLE
[pvr] override SetYear for recordings

### PR DESCRIPTION
**Pull request by phunkyfish**
## Description

Fix issue with year not displaying for PVR recordings introduced by https://github.com/xbmc/xbmc/pull/17217

## Motivation and Context

Alpha3

## How Has This Been Tested?

OSX with pvr.vuplus

## Types of change

- [X] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [X] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [X] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

**Comment by phunkyfish** [comment]
@ksooo when you have some time to review.

**Review by ksooo — APPROVED** [review]
Yep, that's the right fix.

**Comment by emveepee** [comment]
@phunkyfish I have the same comment I made on the other thread.  PVR recordings can't set m_premiered so there is no point in checking it until PVR supports SetPremiereDate()

**Comment by phunkyfish** [comment]
> @phunkyfish I have the same comment I made on the other thread. PVR recordings can't set m_premiered so there is no point in checking it until PVR supports SetPremiereDate()

Can you explain further?

**Comment by phunkyfish** [comment]
Ah, I see what you mean.

**Comment by emveepee** [comment]
> 
> 
> Ah, I see what you mean.

Thinking about it more why not use the new m_year  and worry about m_premiered when it is supported

```

int CPVRRecording::GetYear() const
{
  return m_year;
}

void CPVRRecording::SetYear(int year)
{
  m_year = year;
}


bool CPVRRecording::HasYear() const
{
  return m_year > 0;
}
```

**Comment by phunkyfish** [comment]
I was thinking the same thing.

**Comment by emveepee** [comment]
I think HasYear() is the important test here.

**Comment by phunkyfish** [comment]
> I think HasYear() is the important test here.

Ya, it's late and my brain is tired of fixes!

**Comment by emveepee** [comment]
> 
> 
> > I think HasYear() is the important test here.
> 
> Ya, it's late and my brain is tired of fixes!

You guys have been busy, great job.  The comments for SetYear() now need to change.

File: xbmc/pvr/recordings/PVRRecording.h [hunk 1]
```diff
@@ -339,6 +339,12 @@ namespace PVR
     */
    int GetYear() const override;
 
+   /*!
+    * @brief Set the premiere year of this event.
+    * @param iYear The premiere year
```
Comment (enen92) [inline xbmc/pvr/recordings/PVRRecording.h:344]: Param year and not iYear

**Comment by phunkyfish** [comment]
Hmmm, I'm not so sure.  I will leave it as premiered, less risk. We can refactor this at a later date if we choose.

File: xbmc/pvr/recordings/PVRRecording.h [hunk 2]
```diff
@@ -339,6 +339,12 @@ namespace PVR
     */
    int GetYear() const override;
 
+   /*!
+    * @brief Set the premiere year of this event.
```
Comment (ksooo) [inline xbmc/pvr/recordings/PVRRecording.h:343]: event? did you mean recording?

**Comment by phunkyfish** [comment]
Should be ready now

**Pull request merged**